### PR TITLE
test(sdk) Restored the ParallelFor compiler test data

### PR DESCRIPTION
**Pull request by Ark-kun**
Fixes https://github.com/kubeflow/pipelines/issues/4102

**Description of your changes:**

**Checklist:**
- [x] The title for your pull request (PR) should follow our title convention. [Learn more about the pull request title convention used in this repository](https://github.com/kubeflow/pipelines/blob/master/CONTRIBUTING.md#pull-request-title-convention). 

   PR titles examples:
    * `fix(frontend): fixes empty page. Fixes #1234`
       Use `fix` to indicate that this PR fixes a bug.
    * `feat(backend): configurable service account. Fixes #1234, fixes #1235`
       Use `feat` to indicate that this PR adds a new feature. 
    * `chore: set up changelog generation tools`
       Use `chore` to indicate that this PR makes some changes that users don't need to know.
    * `test: fix CI failure. Part of #1234`
        Use `part of` to indicate that a PR is working on an issue, but shouldn't close the issue when merged.

- [x] Do you want this pull request (PR) cherry-picked into the current release branch?
    
    If yes, use one of the following options:
  
    * **(Recommended.)** Ask the PR approver to add the `cherrypick-approved` label to this PR. The release manager adds this PR to the release branch in a batch update.
    *  After this PR is merged, create a cherry-pick PR to add these changes to the release branch. (For more information about creating a cherry-pick PR, see the [Kubeflow Pipelines release guide](https://github.com/kubeflow/pipelines/blob/master/RELEASE.md#option--git-cherry-pick).)

### CODE REVIEW

**Comment by Ark-kun** [comment]
/retest

I've verified that this PR fixes the issue. But to fully fix the tests, https://github.com/kubeflow/pipelines/pull/4064 need to be merged as well.

@Bobgy can you please take a look?

File: sdk/python/tests/compiler/testdata/parallelfor_item_argument_resolving.yaml [hunk 1]
```diff
@@ -27,36 +27,24 @@ spec:
...
+        image: "python:3.7"
       inputs: 
         parameters: 
           - 
-            name: produce-list-of-strings-output
+            name: produce-list-of-strings-Output
       metadata:
-        annotations:
+        labels:
+          pipelines.kubeflow.org/pipeline-sdk-type: kfp
```
Comment (Bobgy) [inline sdk/python/tests/compiler/testdata/parallelfor_item_argument_resolving.yaml:39]: nit: are these labels telemetry purpose?

**Comment by Bobgy** [comment]
let's wait for #4064 first

**Comment by Bobgy** [comment]
@Ark-kun Can you merge latest master to pick up the travis change, so that we can verify if tests are passing now?

File: sdk/python/tests/compiler/testdata/parallelfor_item_argument_resolving.yaml (hunk 1 above)
Comment (Ark-kun) [reply sdk/python/tests/compiler/testdata/parallelfor_item_argument_resolving.yaml:39]: Hmm. Yes. I'll remove it.

**Comment by Bobgy** [comment]
Looks like two travis tests already passed.
Thanks for the fix @Ark-kun!

/cc @numerology 

/lgtm
/approve

**Comment by numerology** [comment]
Thanks! @Ark-kun

**Pull request merged**